Implement trusted types integration with SVG.

ziransun opened the pull request:
#### 5c7dbebc70ff3f0124a15ca627ff827e8dbdddf1
<pre>
Implement trusted types integration with SVG.
<a href="https://bugs.webkit.org/show_bug.cgi?id=267693">https://bugs.webkit.org/show_bug.cgi?id=267693</a>

Reviewed by Nikolas Zimmermann.

Modify the SVGAnimatedString interface to enforce Trusted Types.
It also split the tests of assignment of SVGScriptElement.href out to a separate test.

* LayoutTests/imported/w3c/web-platform-tests/trusted-types/support/resolve-spv.js: Added.
(promise_spv):
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-svg-script-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-svg-script-set-href-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-svg-script-set-href.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-svg-script.html:
* Source/WebCore/Sources.txt:
* Source/WebCore/svg/SVGAnimatedString.idl:
* Source/WebCore/svg/properties/SVGAnimatedPropertyImpl.h:
* Source/WebCore/svg/properties/SVGAnimatedString.cpp: Added.
(WebCore::SVGAnimatedString::setBaseVal):
* Source/WebCore/svg/properties/SVGAnimatedString.h: Added.
(WebCore::SVGAnimatedString::create):
(WebCore::SVGAnimatedString::SVGAnimatedString):
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/278169@main">https://commits.webkit.org/278169@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/69027b7264d932eb38c502e121842a643bb3a544

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46403 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25545 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48998 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/49078 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42444 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/48710 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29906 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/23030 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37870 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46981 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22582 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40015 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19111 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19951 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/41111 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4448 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42728 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41481 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50907 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21408 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17885 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/45105 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22699 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/40158 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/44522 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10908 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/23085 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22398 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

lukewarlow commented on LayoutTests/imported/w3c/web-platform-tests/trusted-types/WorkerGlobalScope-eval-expected.txt:
```diff
@@ -1,13 +1,8 @@
-CONSOLE MESSAGE: Unrecognized Content-Security-Policy directive 'require-trusted-types-for'.
```
This test is skipped so you can just revert this file change

lukewarlow commented on LayoutTests/imported/w3c/web-platform-tests/trusted-types/WorkerGlobalScope-importScripts-expected.txt:
```diff
@@ -7,14 +7,6 @@ PASS importScripts with two URLs, both strings, in dedicated worker
 PASS importScripts with two URLs, one trusted, in dedicated worker
```
This test is a flaky failure but I would be surprised if these changes are needed? Can probably revert

lukewarlow commented on LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-svg-script-set-href.html:
```diff
@@ -0,0 +1,107 @@
+<!DOCTYPE html>
```
https://github.com/web-platform-tests/wpt/pull/45498

lukewarlow commented on LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-svg-script-set-href-expected.txt:
```diff
@@ -0,0 +1,18 @@
+
+PASS Assign string to SVGScriptElement.href.baseVal.
+PASS Assign TrustedScriptURL to SVGScriptElement.href.baseVal.
+FAIL Assign string to non-attached SVGScriptElement.href via setAttribute. assert_throws_js: function "_ => {
```
These failures are expected due to setAttribute TT implementation not being done.

nikolaszimmermann reviewed (approved):
Looks good, thanks!

ziransun committed:
Implement trusted types integration with SVG.
https://bugs.webkit.org/show_bug.cgi?id=267693

Reviewed by Nikolas Zimmermann.

Modify the SVGAnimatedString interface to enforce Trusted Types.
It also split the tests of assignment of SVGScriptElement.href out to a separate test.

* LayoutTests/imported/w3c/web-platform-tests/trusted-types/support/resolve-spv.js: Added.
(promise_spv):
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-svg-script-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-svg-script-set-href-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-svg-script-set-href.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/trusted-types/trusted-types-svg-script.html:
* Source/WebCore/Sources.txt:
* Source/WebCore/svg/SVGAnimatedString.idl:
* Source/WebCore/svg/properties/SVGAnimatedPropertyImpl.h:
* Source/WebCore/svg/properties/SVGAnimatedString.cpp: Added.
(WebCore::SVGAnimatedString::setBaseVal):
* Source/WebCore/svg/properties/SVGAnimatedString.h: Added.
(WebCore::SVGAnimatedString::create):
(WebCore::SVGAnimatedString::SVGAnimatedString):
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/278169@main
